### DISCUSSION TITLE
Gridvania Translation

### DISCUSSION THREAD

**Discussion by Giesch** [Trouv/bevy_ecs_ldtk#198] [Q&A] [answered]
Is there a way to query for an LDtk entity's world translation? IE, not GlobalTransform, but something like a GridVaniaTransform?

I'm using the Added<Component> filter pattern to add physics components to newly-spawned LDtk entities. I just started adding levels past the first, with a GridVania layout. When querying for transforms of added tiles, I'm getting their in-level transform, so all components added in a system using the Added<Component> filter spawn in the 'right' place in the first level, and then the surrounding levels have sprites, but no late-added components. This is with load_level_neighbors: true; if that's false, then only the first level's additional components get added, but the surrounding levels never have their additional components added.

I assume I'm doing something wrong, but I don't know what the right way to go about this is. Should I be doing my own Translation offsets by multiples of the worldX and worldY of the appropriate level?

Thanks!

**Comment by Trouv** [comment, marked as answer]
Hmm I'm not completely sure I understand the problem.

> When querying for transforms of added tiles, I'm getting their in-level transform, so all components added in a system using the Added filter spawn in the 'right' place in the first level, and then the surrounding levels have sprites, but no late-added components.

So the components intended for entites that aren't in the first level are still appearing in the first level? I have a couple guesses for what the problem could be..
1. The physics engine you're using only pays attention to `Transform` and not `GlobalTransform`. We used to have this problem with rapier but we don't anymore.
2. You're not actually adding these components to the entities that the plugin spawned, instead trying to create new entities with the same transform. You should probably be adding the late components to the entity the plugin spawned. If this is your problem, you can add components to existing entities with bevy `Commands` via `commands.entity(existing_entity).insert(...)`. If you're adding an entire bundle to the existing entity, you may need to be careful not to overwrite the transform provided by the plugin.

I hope this helps, let me know if I misunderstood the issue or misdiagnosed it.

**Comment by Giesch** [comment]
It's almost certainly a variation on 1. I'm doing custom physics as a learning exercise and may be doing other things wrong as well. Thanks for pointing it out!
